python/lbann/modules/pytorch.py

```diff
@@ -0,0 +1,69 @@
+import numpy as np
+
+import lbann
+import lbann.modules
+from lbann.util import str_list
+
+# Mimics torch.matmul in LBANN
+def PytorchMatmul(x, x_shape, y, y_shape, return_dims=False):
+    if len(x_shape) != len(y_shape):
+        raise RuntimeError(
+            "Broadcasting not fully implemented, tensors must have same dimension"
+        )
+    need_reshape = (len(x_shape) > 3) and (len(y_shape) > 3)
+    if need_reshape:
+        if x_shape[:-2] != y_shape[:-2]:
+            raise RuntimeError("The first n-2 dimensions must match")
+        new_x_shape = (np.prod(x_shape[:-2]),) + x_shape[-2:]
+        x = lbann.Reshape(x, dims=str_list(new_x_shape))
+
+        new_y_shape = (np.prod(y_shape[:-2]),) + y_shape[-2:]
+        y = lbann.Reshape(y, dims=str_list(new_y_shape))
+
+    z = lbann.MatMul(x, y)
+
+    z_shape = x_shape[:-1] + (y_shape[-1],)
+    if need_reshape:
+        z = lbann.Reshape(z, dims=str_list(z_shape))
+
+    if return_dims:
+        return z, z_shape
+    return z
+
+
+# Mimics torch.nn.Linear in LBANN
+def PytorchLinear(x, input_shape, hidden_size, weights=[], name="", return_dims=False):
+    need_reshape = len(input_shape) > 2
```

Comment:
I think we need a reshape if the data is 1D, or we should use the basic fully-connected layer.

Reply:
Added a check for 1D inputs with recent commit.